data/statuses.js

```diff
@@ -523,7 +523,8 @@ let BattleStatuses = {
 		onTryMove: function (target, source, effect) {
 			if (effect.type === 'Fire' && effect.category !== 'Status') {
 				this.debug('Primordial Sea fire suppress');
-				this.add('-fail', source, effect, '[from] Primordial Sea');
+				this.add('-fail', '', effect, '[from] Primordial Sea');
```

Comment:
It looks like the in-game message is "The Fire-type attack fizzled out in the heavy rain!"

Why are you hiding the attacker? The attacker's identity is known:

![image](https://user-images.githubusercontent.com/551184/47816260-a44cf900-dd20-11e8-8645-7b92f81b579e.png)

Reply:
The parameters are named incorrectly. The 2nd is really the target https://github.com/Zarel/Pokemon-Showdown/blob/master/data/scripts.js#L224

Reply:
Oh, you can just rename the parameters.

```suggestion
				this.add('-fail', attacker, effect, '[from] Primordial Sea');
```